### ISSUE TITLE
UndefinedFunctionError with ecto 3.8.1

### ISSUE THREAD

**Issue by MrYawe** [evadne/etso#19]
After upgrading `ecto_sql` and `ecto` from `3.7.1` to `3.8.1` I have the following error :

```elixir
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]** (stop) {:EXIT, {%UndefinedFunctionError{arity: 2, function: :fetch, message: nil, module: Etso.Adapter.Meta, reason: "Etso.Adapter.Meta does not implement the Access behaviour"}, [{Etso.Adapter.Meta, :fetch, [%{__struct__: Etso.Adapter.Meta, adapter: Etso.Adapter, cache: #Reference<0.604452022.765329409.233143>, pid: #PID<0.4743.0>, repo: LogflareLogger.Repo}, :stacktrace], []}, {Access, :get, 3, [file: 'lib/access.ex', line: 285]}, {Ecto.Repo.Supervisor, :tuplet, 2, [file: 'lib/ecto/repo/supervisor.ex', line: 161]}, {LogflareLogger.Repo, :insert!, 2, [file: 'lib/repo.ex', line: 2]}, {LogflareLogger.BatchCache, :put, 2, [file: 'lib/logflare_logger/batch_cache.ex', line: 17]}, {LogflareLogger.HttpBackend, :handle_event, 2, [file: 'lib/logflare_logger/http_backend.ex', line: 41]}, {:gen_event, :server_update, 4, [file: 'gen_event.erl', line: 627]}, {:gen_event, :server_notify, 4, [file: 'gen_event.erl', line: 609]}]}}
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]Last message: {:gen_event_EXIT, LogflareLogger.HttpBackend, {:EXIT, {%UndefinedFunctionError{arity: 2, function: :fetch, message: nil, module: Etso.Adapter.Meta, reason: "Etso.Adapter.Meta does not implement the Access behaviour"}, [{Etso.Adapter.Meta, :fetch, [%{__struct__: Etso.Adapter.Meta, adapter: Etso.Adapter, cache: #Reference<0.604452022.765329409.233143>, pid: #PID<0.4743.0>, repo: LogflareLogger.Repo}, :stacktrace], []}, {Access, :get, 3, [file: 'lib/access.ex', line: 285]}, {Ecto.Repo.Supervisor, :tuplet, 2, [file: 'lib/ecto/repo/supervisor.ex', line: 161]}, {LogflareLogger.Repo, :insert!, 2, [file: 'lib/repo.ex', line: 2]}, {LogflareLogger.BatchCache, :put, 2, [file: 'lib/logflare_logger/batch_cache.ex', line: 17]}, {LogflareLogger.HttpBackend, :handle_event, 2, [file: 'lib/logflare_logger/http_backend.ex', line: 41]}, {:gen_event, :server_update, 4, [file: 'gen_event.erl', line: 627]}, {:gen_event, :server_notify, 4, [file: 'gen_event.erl', line: 609]}]}}}
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]11:42:12.527 [error] :gen_event handler LogflareLogger.HttpBackend installed in Logger terminating
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]** (UndefinedFunctionError) function Etso.Adapter.Meta.fetch/2 is undefined (Etso.Adapter.Meta does not implement the Access behaviour)
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (etso 0.1.6) Etso.Adapter.Meta.fetch(%{__struct__: Etso.Adapter.Meta, adapter: Etso.Adapter, cache: #Reference<0.604452022.765329409.233143>, pid: #PID<0.4743.0>, repo: LogflareLogger.Repo}, :stacktrace)
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (elixir 1.13.4) lib/access.ex:285: Access.get/3
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (ecto 3.8.1) lib/ecto/repo/supervisor.ex:161: Ecto.Repo.Supervisor.tuplet/2
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (logflare_logger_backend 0.11.0) lib/repo.ex:2: LogflareLogger.Repo.all/2
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (logflare_logger_backend 0.11.0) lib/logflare_logger/batch_cache.ex:103: LogflareLogger.BatchCache.events_in_flight/0
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (logflare_logger_backend 0.11.0) lib/logflare_logger/http_backend.ex:56: LogflareLogger.HttpBackend.handle_info/2
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (stdlib 3.17.2) gen_event.erl:627: :gen_event.server_update/4
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]    (stdlib 3.17.2) gen_event.erl:609: :gen_event.server_notify/4
2022-05-05T11:42:12Z app[e71f06cd] cdg [info]Last message: :in_flight_check
```

**Comment by evadne** [comment]
Given `Ecto.Adapter.adapter_meta` typedoc — “It must be a map” and Ecto always sets `:cache` and `:pid`, the proper fix would be to add cache & pid keys to the struct and to implement Access via delegation to Map

IMO

**Comment by evadne** [comment]
@MrYawe are you using a fork of etso.

**Comment by evadne** [comment]
Given https://github.com/elixir-ecto/ecto/blob/master/CHANGELOG.md#bug-fix this seems like a regression in Ecto which was fixed in 3.8.2 so will close this issue for now. However will update Etso for general Ecto 3.8 compatibility.
